import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/cstore/multipledest/MultipleDestinationUploadService.java

```diff
@@ -0,0 +1,115 @@
+package com.google.cloud.healthcare.imaging.dicomadapter.cstore.multipledest;
+
+import com.google.cloud.healthcare.IDicomWebClient;
+import com.google.cloud.healthcare.imaging.dicomadapter.AetDictionary;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.BackupState;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.IBackupUploadService;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.IBackupUploader.BackupException;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.multipledest.sender.CStoreSender;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.multipledest.sender.CStoreSenderFactory;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+public class MultipleDestinationUploadService implements IMultipleDestinationUploadService {
+
+  private Logger log = LoggerFactory.getLogger(MultipleDestinationUploadService.class);
+
+  private CStoreSenderFactory cStoreSenderFactory;
+  private IBackupUploadService backupUploadService;
+  private int attemptsAmount;
+
+  public MultipleDestinationUploadService(CStoreSenderFactory cStoreSenderFactory, IBackupUploadService backupUploadService, int attemptsAmount) {
+    this.cStoreSenderFactory = cStoreSenderFactory;
+    this.backupUploadService = backupUploadService;
+    this.attemptsAmount = attemptsAmount;
+  }
+
+  @Override
+  public void start(ImmutableList<IDicomWebClient> healthcareDestinations,
+                    ImmutableList<AetDictionary.Aet> dicomDestinations,
+                    InputStream inputStream,
+                    String sopClassUID,
+                    String sopInstanceUID) throws MultipleDestinationUploadServiceException {
+    CStoreSender cStoreSender = cStoreSenderFactory.create();
```

Comment:
Could we initialize this only if len(dicomDestinations) > 0 to save on resources / connections?

Reply:
it`s only Factory class instance creation. Connection will be created on create() method call. Check len(dicomDestinations) > 0 adde to MultipleDestinationUploadService. Thank you.